### ISSUE TITLE
trying to debug an rcloud function goes into an infinite loop

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#1310]
of course, but mostly wanted to point out this is possible.

the session pane does not have an input control, so when the console.in comes in, there is no cell to handle it, and the R side attempts to parse the error.

and, loop.

**Comment by s-u** [comment]
Ideally, there should be a place for cell-less input since technically R can issue ReadConsole requests even outside of cell execution.

**Comment by gordonwoodhull** [comment]
the problem here is that if we add an input to the session pane then everyone will want to just use the session pane as a console, and we'll lose reproducibility again.  :-(

**Comment by s-u** [comment]
This would be just a fall-back in case no cell is active, so normally users would never see it, but it would save us in the odd cases where we run into infinite loops due to missing input.

**Comment by gordonwoodhull** [comment]
a hidden feature.  i like that.  will turn into an enhancement request and implement for 1.4 (unless my next 36 hours open up magically, in which case i'll do it sooner).  

it would actually be super helpful for debugging rcloud itself, at least until we enable compute separation - i was trying to decipher `rcloud.search` so I thought I would give it a try.

**Comment by gordonwoodhull** [comment]
man, this would sure be helpful. it's a pity that it's a week of work that i can't spare.

**Comment by gordonwoodhull** [comment]
Of course, there are special problems with this and compute separation. It's a security hole (so we'd need admin access or debug mode), and I'm not sure if Rserve/RCloud code is expecting the control process to request input?

**Comment by s-u** [comment]
This would still be on compute, not control. We never ever want any user input to control.

**Comment by gordonwoodhull** [comment]
So this is of limited usefulness. It would basically only help with debugging callbacks from interactive cells, not the core RCloud code I was imagining.
